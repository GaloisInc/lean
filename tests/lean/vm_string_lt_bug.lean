open native

#eval to_bool ("a" < "b")
#eval to_bool ("b" < "b")

namespace test1
meta def m := rb_map.mk string nat

meta def m' := m.insert "foo" 10

#eval m'.find "foo"
end test1

namespace test2

meta def m := rb_map.mk nat nat

meta def m' := m.insert 3 10

#eval m'.find 3

end test2
