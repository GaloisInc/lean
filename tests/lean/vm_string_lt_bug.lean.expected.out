tt
ff
(some 10)
(some 10)
